apps/whale-api/src/module.indexer/model/dftx/update.masternode.spec.ts

```diff
@@ -0,0 +1,184 @@
+import { createTestingApp, DelayedEunosPayaTestContainer, invalidateFromHeight, stopTestingApp, waitForIndexedHeight } from '../../../e2e.module'
+import { NestFastifyApplication } from '@nestjs/platform-fastify'
+import { JsonRpcClient } from '@defichain/jellyfish-api-jsonrpc'
+import { RegTest } from '@defichain/jellyfish-network'
+import { P2WPKH } from '@defichain/jellyfish-address'
+import { MasternodeMapper } from '../../../module.model/masternode'
+import { MasternodeStatsMapper } from '../../../module.model/masternode.stats'
+
+describe('Update masternode', () => {
+  const container = new DelayedEunosPayaTestContainer()
+  let app: NestFastifyApplication
+  let client: JsonRpcClient
+  let masternodeId: string
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
+
+    app = await createTestingApp(container)
+    client = new JsonRpcClient(await container.getCachedRpcUrl())
+
+    await container.generate(1)
+
+    const ownerAddress = await client.wallet.getNewAddress()
+    masternodeId = await client.masternode.createMasternode(ownerAddress)
+    await container.generate(20)
+
+    const height = await client.blockchain.getBlockCount()
+    // enable updating
+    await client.masternode.setGov({
+      ATTRIBUTES: {
+        'v0/params/feature/mn-setowneraddress': 'true',
+        'v0/params/feature/mn-setoperatoraddress': 'true',
+        'v0/params/feature/mn-setrewardaddress': 'true'
+      }
+    })
+    await container.generate(70)
+    await waitForIndexedHeight(app, height)
+  })
+
+  afterAll(async () => {
+    await stopTestingApp(container, app)
+  })
+
+  it('should index update operatorAddress, ownerAddress, and rewardAddress in masternode', async () => {
+    const masternodeMapper = app.get(MasternodeMapper)
+    const masternode = await masternodeMapper.get(masternodeId)
+    expect(masternode).not.toStrictEqual(undefined)
+
+    const address = await container.getNewAddress('', 'bech32')
+    const addressDest: P2WPKH = P2WPKH.fromAddress(RegTest, address, P2WPKH)
+
+    await client.masternode.updateMasternode(masternodeId, {
+      operatorAddress: addressDest.utf8String,
+      rewardAddress: addressDest.utf8String
+      // ownerAddress: addressDest.utf8String // @TODO(kvenho) Blockchain is not fix yet
+    })
+
+    await container.generate(1)
+    const updateHeight = await client.blockchain.getBlockCount()
+    await container.generate(1)
+    await waitForIndexedHeight(app, updateHeight)
+
+    const gotMasternodeForReward = await client.masternode.getMasternode(masternodeId)
+    // expect(gotMasternodeForReward[masternodeId]?.ownerAuthAddress).toStrictEqual(addressDest.utf8String) // @TODO(kvenho) Blockchain is not fix yet
+    expect(gotMasternodeForReward[masternodeId]?.operatorAuthAddress).toStrictEqual(addressDest.utf8String)
+    expect(gotMasternodeForReward[masternodeId]?.rewardAddress).toStrictEqual(addressDest.utf8String)
```

Comment:
Just realise. I think testing value one by one is correct way to do, because if we write the test with ONE `toStrictEqual`, we need to include the WHOLE object. E.g.

```
expect(mnResult[masternodeId]).toStrictEqual({
      ownerAuthAddress: addressDest.utf8String,
      operatorAuthAddress: addressDest.utf8String,
      rewardAddress: addressDest.utf8String,
      creationHeight: 103,
      resignHeight: -1,
      resignTx: '0000000000000000000000000000000000000000000000000000000000000000',
      collateralTx: 'd1b9bfbf5a7a1beab499be8bb05654f61398e305a942ac6a1ff0362d8ed31b34', <-- And, this is different in every test
      state: 'ENABLED',
      mintedBlocks: 0,
      ownerIsMine: true,
      operatorIsMine: true,
      localMasternode: false,
      targetMultipliers: [1, 1]
    })
```

Reply:
If the collateral txid cannot be obtained deterministically, then just assert the type.
```js
expect(mnResult[masternodeId]).toStrictEqual({
      ownerAuthAddress: addressDest.utf8String,
      operatorAuthAddress: addressDest.utf8String,
      rewardAddress: addressDest.utf8String,
      creationHeight: 103,
      resignHeight: -1,
      resignTx: '0000000000000000000000000000000000000000000000000000000000000000',
      collateralTx: expect.stringMatching(/[0-f]{64}/),
      state: 'ENABLED',
      mintedBlocks: 0,
      ownerIsMine: true,
      operatorIsMine: true,
      localMasternode: false,
      targetMultipliers: [1, 1]
    })
```